cirq/ops/raw_types.py

```diff
@@ -392,6 +393,24 @@ def with_qubits(self, *new_qubits: 'cirq.Qid') -> 'cirq.Operation':
                 `qubits` property.
         """
 
+    def with_tags(self, *new_tags: Any) -> 'cirq.TaggedOperation':
```

Comment:
Should `TaggedOperation` override this to add `new_tags` to itself (or replace its tags with `new_tags`), instead of nesting it in another `TaggedOperation`?

Reply:
Good idea.  I added a with_tags() on TaggedOperation that avoids nesting.